internal/host/common.go

```diff
@@ -191,6 +192,22 @@ func updateRole(h *models.Host, role models.HostRole, db *gorm.DB, srcRole *stri
 		return errors.Wrapf(updateReply.Error, "failed to update host %s from cluster %s role to %s",
 			h.ID.String(), h.ClusterID.String(), role)
 	}
+
+	if IsDay2Host(h) {
```

Comment:
I don't think this code should be here: 
1) currently day2 supports only workers
2) this can cause machine_config_pool to be updated when user/UI changes the role without changing config_pool.

Reply:
1. That's right, but with this code it would support also masters.
2. I can update to
```golang
if IsDay2Host(h) && (h.MachineConfigPoolName == "" || h.MachineConfigPoolName == *srcRole)
```

The proper way is that machineConfigPool would equal the host role unless asking differently.

Reply:
yes, let's add the condition that you suggested